tls/src/main/java/org/bouncycastle/jsse/provider/CipherSuiteInfo.java

```diff
@@ -210,6 +210,9 @@ private static void decomposeKeyExchangeAlgorithm(Set<String> decomposition, int
         case KeyExchangeAlgorithm.RSA:
             addAll(decomposition, "RSA");
             break;
+        case KeyExchangeAlgorithm.DH_anon:
+        case KeyExchangeAlgorithm.ECDH_anon:
+            break;
```

Comment:
See here: https://github.com/openjdk/jdk/blob/8567266795cd1171f5b353d0e0813e7eeff319c2/src/java.base/share/classes/sun/security/ssl/SSLAlgorithmDecomposer.java#L94-L101 .

In our code this would correspond to adding entries to decompositionTLS. If done right, then when the security property 'jdk.tls.disabledAlgorithms' contains 'anon' as one of its entries (which is the case by default in current JDKs) then none of these anon suites should be enabled (i.e. they get effectively disabled). Removing 'anon' from the security property should then re-enable them (subject to other constraints).